ui/src/plugins/dev.perfetto.AndroidLog/logs_panel.ts

```diff
@@ -61,6 +73,20 @@ interface LogEntries {
   totalEvents: number; // Count of the total number of events within this window
 }
 
+async function getMachineIds(engine: Engine): Promise<number[]> {
+  // A machine might not provide Android logs, even if configured to do so.
+  // Hence, the |cpu| table might have ids not present in the logs. Given this
+  // is highly unlikely and going through all logs is expensive, we will get
+  // the ids from |cpu|, even if filter shows ids not present in logs.
+  const result = await engine.query(`SELECT DISTINCT(machine_id) FROM cpu`);
+  const machineIds: number[] = [];
+  const it = result.iter({machine_id: NUM_NULL});
+  for (let row = 0; it.valid(); it.next(), row++) {
+    machineIds.push(it.machine_id == null ? 0 : it.machine_id);
```

Comment:
I'd suggest `it.machine_id ?? 0`.

Generally `==` should be avoided in our codebase.